tensorflow/core/kernels/tensor_forest/leaf_model.h

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CORE_KERNELS_TENSOR_FOREST_LEAF_MODEL_H_
+#define TENSORFLOW_CORE_KERNELS_TENSOR_FOREST_LEAF_MODEL_H_
+
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/kernels/tensor_forest/tensor_forest.pb.h"
+
+namespace tensorflow {
+
+using tensorforest::Leaf;
+using tensorforest::LeafStat;
+
+typedef TTypes<const float, 2>::ConstTensor DenseTensorType;
+
+namespace {
+enum LeafModelType { CLASSIFICATION = 0, REGRESSION = 1 };
+}
+
+class LeafModelOperator {
```

Comment:
Do we need this abstraction at all?

Reply:
there is a difference for leafs when `ExportModel` and `UpdateModel` between classification https://github.com/tensorflow/tensorflow/pull/21579/files#diff-c000cbb3c7359ba2e87d29b7b3e4c399R63 and regression https://github.com/tensorflow/tensorflow/pull/21579/files#diff-c000cbb3c7359ba2e87d29b7b3e4c399R70.  

I think the idea is for every protobuf object, we have a operator class equivalent for it.  

But still, what would you suggest if we remove this abstraction?